frontend/src/metabase/models/containers/ModelDetailPage/ModelDetailPage.unit.spec.tsx

```diff
@@ -465,6 +465,28 @@ describe("ModelDetailPage", () => {
           ).toBeInTheDocument();
         });
 
+        it("lists existing public query actions with public label", async () => {
+          const model = getModel();
+          const action = createMockQueryAction({
+            model_id: model.id(),
+            public_uuid: "mock-uuid",
+          });
+          await setupActions({ model, actions: [action] });
+
+          expect(screen.getByText(action.name)).toBeInTheDocument();
+          expect(screen.getByText(TEST_QUERY)).toBeInTheDocument();
+          expect(screen.getByText(/Public Action/)).toBeInTheDocument();
+          const expectedSeparator = "·";
+          expect(
+            screen.getByText(new RegExp(expectedSeparator)),
+          ).toBeInTheDocument();
+          expect(
+            screen.getByText(
+              new RegExp(`Created by ${action.creator.common_name}`),
+            ),
+          ).toBeInTheDocument();
```

Comment:
This we're testing this already?

Reply:
I mean, it's possible that we somehow only show `Public action` but not the creator name because I implemented it badly. So we should check that both `public action` and the creator name are still present.

Reply:
TBH that shouldn't be the case. The only reason the creator is optional is that we added the creator column after we added the action table. Maybe we should just mark it as required, so we always assume the creator is there. It can only fail if you have some old actions locally, but that'd be fair to delete them vs. increasing code complexity. WDYT? We shouldn't necessarily fix this in that PR though

Reply:
Hmm how do I phrase this? I understand your point, but E2E for me should be unaware of its implementation and only test what we're seeing on the screen.

From the spec we know that there are 2 possible scenarios for subtitles:
1. `Created by John Wick`
2. `Public Action · Created by John Wick`

We're not talking about the case in development where it could just show `Public Action` without the creator name.

Now I'm talking about the test itself. 1. is covered by an existing test. But 2. is what I'm adding here. Now here's the question. Should we just only assert `Public Action` in the second test? If each test is independent that means the test I'm adding must be meaningful by its own, and I'm not seeing how only asserting `Public Action` would help that. It's more clear to me if we assert both `Public Action` and the creator name, and it doesn't have anything to do with any other test.

Back to my original comment. It's possible that I produce the code that passes these 2 tests
1. test the creator name exist
2. test if `Public Action` exist

but in reality, the code could produce subtitles like this (despite every action having a `creator` property):
1. `Created by John Wick`
2. `Public Action`

Our tests would be green but this feature is implemented incorrectly.

Do I misunderstand anything? Hope this clear things up.

Reply:
Nope, you got that right, and I'm overthinking this. Agreeing we should keep that assertion. I just started to think more about our optional creator checks in code vs. the fact that we just want to make sure we display it